cyclonedx-bom/src/models/bom.rs

```diff
@@ -73,10 +77,13 @@ impl Bom {
         self,
         writer: &mut W,
     ) -> Result<(), crate::errors::JsonWriteError> {
-        let bom: crate::specs::v1_3::bom::Bom = self.into();
+        let bom: crate::specs::v1_3::bom::Bom = self
+            .try_into()
+            .expect("todo: error handling in output_as_json_v1_3");
```

Comment:
Another unfulfilled TODO

Reply:
I've looked into this. I had to switch from into() to try_into() because of the different handling of `version` in `Component` in 1.3 vs. 1.4. In 1.3 it is mandatory, in 1.4 is is optional. So, a conversion into 1.3 might fail when the `Component` has no  `version`.
The thing is that `try_into` returns either `Ok(Bom)` or `Error(BomError)`, while `output_as_json_v1_3()` returns a `Ok(())` or a `Error(JsonWriteError)` but not a `Error(BomError)`.

So, how can we handle this. I see three options:

1. We do not change the interface of `output_as_json_v1_3()` and if a `BomError` occurs we write the error as a JSON document.
2. We do not change the interface of `output_as_json_v1_3()` and if a `BomError` occurs we translate the `BomError` somehow into a `JsonWriteError`.
3. We change the interface of `output_as_json_v1_3()`, which would introduce a breaking change.

We face the same challenge for the `output_as_xml_v1_3()`. The `output_as_*_v1_4()` are not affected because there we are able to use `into()`. However, we could also change it to `try_into()` to align it.

Are there other options I'm not aware of and which approach should we follow? Please advice. Thanks.

Reply:
We are making a field optional to match the spec, which is going to introduce breaking changes no matter if it is `pub` or not. I think it's a good idea to change the signature to reflect reality while we're making breaking changes anyway.

I've looked at the types and `BomError` represents the possible failure modes better than a hypothetical modification of `JsonWriteError`, so let's go with option 3.

Reply:
Ok, great. I'll follow this approach. Thanks.